data-avro/src/test/java/com/linkedin/data/avro/TestSchemaTranslator.java

```diff
@@ -75,6 +75,162 @@ public void testTranslateDefaultBackwardsCompatibility()
+                    "            \"type\": [" +
+                    "              \"null\"," +
+                    "              \"int\"" +
+                    "            ]," +
+                    "            \"default\": null" +
+                    "          }," +
+                    "          {" +
+                    "            \"name\": \"a2\"," +
+                    "            \"type\": [" +
+                    "              \"null\"," +
+                    "              \"string\"" +
+                    "            ]," +
+                    "            \"default\": null" +
+                    "          }," +
+                    "          {" +
+                    "            \"name\": \"fieldDiscriminator\"," +
+                    "            \"type\": {" +
+                    "              \"type\": \"enum\"," +
+                    "              \"name\": \"testUnionWithAliasFieldDiscriminator\"," +
+                    "              \"symbols\": [" +
+                    "                \"a1\"," +
+                    "                \"a2\"" +
+                    "              ]" +
+                    "            }," +
+                    "            \"doc\": \"Contains the name of the field that has its value set.\"" +
+                    "          }" +
+                    "        ]" +
+                    "      }," +
+                    "      \"customAnnotation\": {" +
+                    "        \"/annotationKey\": \"annotationValue\"" +
+                    "      }" +
+                    "    }" +
+                    "  ]" +
+                    "}"
+            },
+             // Test : field properties merged with Typeref properties
+            {
+                "record test {" +
+                    "  @compliance.`/fieldDiscriminator` = \"NONE\" " +
+                    "  unionTyperef:" +
+                    "  @compliance = {" +
+                    "    \"/string\": \"NONE\"" +
+                    "  }" +
+                    "  typeref unionRefNoAlias =" +
+                    "  union[int, string]" +
+                    "}"
+              ,
+                "{" +
+                    "  \"type\": \"record\"," +
+                    "  \"name\": \"test\"," +
+                    "  \"fields\": [" +
+                    "    {" +
+                    "      \"name\": \"unionTyperef\"," +
+                    "      \"type\": [" +
+                    "        \"int\"," +
+                    "        \"string\"" +
+                    "      ]," +
+                    "      \"compliance\": {" +
+                    "        \"/fieldDiscriminator\": \"NONE\"," +
+                    "        \"/string\": \"NONE\"" +
+                    "      }" +
+                    "    }" +
+                    "  ]" +
+                    "}"
+            },
+            // Test : field properties overrides Typeref properties
```

Comment:
You should add a test for the case where the field specifies properties for a union member and another where it specifies property for "fieldDiscriminator" field. The test should check that the properties are propagated correctly to the nested fields representing the members.

Reply:
This should be for union with alias.

Reply:
@karthikbalasub I don't think we are propagating the `"on-the-field" union member annotation` to `nested fields` in the translator. Isn't that the job of annotation processor?

Reply:
@karthikbalasub 
> This should be for union with alias.
You are right